internal/provider/resource_port_forward.go

```diff
@@ -0,0 +1,180 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/paultyng/go-unifi/unifi"
+)
+
+//	GetPortForward(site string, id string) (*unifi.PortForward, error)
+//	DeletePortForward(site string, id string) (*unifi.PortForward, error)
+//	CreatePortForward(site string, d *unifi.PortForward) (*unifi.PortForward, error)
+//	UpdatePortForward(site string, d *unifi.PortForward) (*unifi.PortForward, error)
+
+func resourcePortForward() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePortForwardCreate,
+		Read:   resourcePortForwardRead,
+		Update: resourcePortForwardUpdate,
+		Delete: resourcePortForwardDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"dst_port": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(portForwardDstPortRegexp, "dst_port is invalid"),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"fwd": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(portForwardFwdRegexp, "fwd is invalid"),
+			},
+			"fwd_port": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(portForwardFwdPortRegexp, "fwd_port is invalid"),
+			},
+			"log": {
+				Type:     schema.TypeBool,
+				Default:  false,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+			},
+			"port_forward_interface": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"wan", "wan2", "both", ""}, false),
+			},
+			"proto": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "tcp_udp",
+				ValidateFunc: validation.StringInSlice([]string{"tcp_udp", "tcp", "udp"}, false),
+			},
+			"src": {
+				Type:         schema.TypeString,
+				Required:     false,
+				Optional:     true,
+				Default:      "any",
+				ValidateFunc: validation.StringMatch(portForwardSrcRegexp, "src is invalid"),
+			},
+		},
+	}
+}
+
+func resourcePortForwardCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*client)
+
+	req, err := resourcePortForwardGetResourceData(d)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.c.CreatePortForward(c.site, req)
+	if err != nil {
+		apiErr, ok := err.(*unifi.APIError)
+		if !ok {
+
+			return fmt.Errorf("ResourcePortForwardCreate api error %w. With %w", apiErr, err)
+		}
+
```

Comment:
This is leaking errors, probably just simplest to return the raw error:
```suggestion
		return err
```

Reply:
I copied this code from the user.go file where you caught the apiError and did some magic with it, wasn't sure if this would be required here, if an error occured. If you don't see an issue, I'll remove it

Reply:
Yeah, the user has a special flow for existing MAC's when registering, wouldn't apply here.